spring-xd-dirt/src/main/java/org/springframework/xd/dirt/cluster/ContainerFilter.java

```diff
@@ -28,6 +28,6 @@
  */
 public interface ContainerFilter {
 
-	List<Container> filterContainers(ModuleDescriptor moduleDescriptor, Iterable<Container> containers);
+	Collection<Container> filterContainers(ModuleDescriptor moduleDescriptor, Iterable<Container> containers);
```

Comment:
Right, having the entire collection of containers is more flexible. What do you think the return type should be?